client.go

```diff
@@ -164,6 +168,22 @@ func (c *Client) OnConnected(f OnConnectedFunc) {
 	c.onConnectedFuncs = append(c.onConnectedFuncs, f)
 }
 
+// OnError can be used to hook into the errors that the client encounters
+// outside the normal request/response cycle. Note that it is not possible to
+// call [Client.Stop] inside f. If you want to stop the client from here, you
+// should use a goroutine. This is because [Client.Stop] will block until the
+// client has stopped.
+func (c *Client) OnError(f OnErrorFunc) {
```

Comment:
I wonder why we didn't make `OnConnected` chainable like the rest. Should we follow that path or just this one?

Also wondering if we should accept variadict input if you have multiple ones, then it would at least be a single call if you for some reason have multiple ones.

Reply:
That is a great question! There is time to change that since we haven't released the next version yet.